pylint/reporters/multi_reporter.py

```diff
@@ -88,15 +89,15 @@ def display_messages(self, layout: BaseLayout) -> None:
         for rep in self._sub_reporters:
             rep.display_messages(layout)
 
-    def on_set_current_module(self, module: str, filepath: Optional[AnyPath]) -> None:
+    def on_set_current_module(self, module: str, filepath: PosixPath) -> None:
```

Comment:
This is based on what the current test suite does. Apparently this function never gets called with `None` or `str`.
However, I think it would be better to change this indeed. I'm going to change this to `Optional[PosixPath, str]` and do a separate commit in which I do `Union[None, PosixPath, str]` -> `Optional[PosixPath, str]` for all other cases.